go-controller/pkg/ovn/policy.go

```diff
@@ -904,6 +902,18 @@ func (oc *Controller) addNetworkPolicy(policy *knet.NetworkPolicy) {
 				handler.gress, np)
 		}
 	}
+
+	// Finally, make sure that all ACLs are set
+	np.Lock()
+	if !np.deleted {
+		for _, gp := range np.ingressPolicies {
+			gp.localPodSetACL(np.portGroupName, np.portGroupUUID, nsInfo.aclLogging.Allow)
+		}
+		for _, gp := range np.egressPolicies {
+			gp.localPodSetACL(np.portGroupName, np.portGroupUUID, nsInfo.aclLogging.Allow)
+		}
+	}
+	np.Unlock()
```

Comment:
I understand this is style? I will change it to your suggestion but I don't usually use that style unless I know the lock needs to be held until the end of the scope for sure. In this case the scope is the function and in that function the lock is held sometimes but not always, so that means that the function itself does not logically need to hold the lock at all times. For example, some one might add additional code on that function after that section that does not need to hold the lock but might miss the defer and keep the lock inadvertently.

Reply:
we try to never manually unlock and always use defer. If the logic requires you to unlock before the end of the function then the logic probably belongs in its own function where you can use defer. Makes it so we never miss unlocking somewhere further down the code.

Reply:
I set this up on its own function, let's see how that looks.